### PR TITLE
Add history resolution and use prometheus client

### PR DESCRIPTION
**Pull request by yashbhutwala**
Follow-up from PR #2549; only includes:
- [X] add history resolution
- [X] use official prometheus client; remove previous client

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/recommender/input/history/history_provider.go [hunk 1]
```diff
@@ -17,23 +17,29 @@ limitations under the License.
...
 
 import (
+	"context"
 	"fmt"
-	"net/http"
 	"sort"
 	"strings"
 	"time"
 
+	promapi "github.com/prometheus/client_golang/api"
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/recommender/input/history/history_provider.go:26]: nit: I think it's cleaner to have the github.com imports in one group and the k8s.io/autoscaler ones in the other.

File: vertical-pod-autoscaler/pkg/recommender/input/history/history_provider.go [hunk 2]
```diff
@@ -58,19 +64,44 @@ type HistoryProvider interface {
...
+		Address: config.Address,
+	})
+	if err != nil {
+		return &prometheusHistoryProvider{}, err
+	}
+
+	// Use Prometheus's model.Duration; this can additionally parse durations in days, weeks and years (as well as seconds, minutes, hours etc)
+	historyDuration, err := prommodel.ParseDuration(config.HistoryLength)
+	if err != nil {
+		return &prometheusHistoryProvider{}, fmt.Errorf("history length %s is not a valid duration: %v", config.HistoryLength, err)
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/recommender/input/history/history_provider.go:86]: nit: valid Prometheus duration
(here and below)

**Comment by yashbhutwala** [comment]
@bskiba Thank you for the review!  I fixed the nits, as well as tested this manually (image of vpa-recommender used [here](https://hub.docker.com/layers/ybhutdocker/vpa-recommender/history-resolution-6/images/sha256-b2e67533868c539f781c3d998f7a13c50bd062e2c233eefce91d5330229a1675?context=repo)), I had to fix the promql query [here](https://github.com/kubernetes/autoscaler/pull/3122/commits/846c656bbd1df92e2df173b44970ba6fe259100f).  Let me know if you are OK with the changes now 😄

**Comment by cliles** [comment]
@bskiba @jbartosik @krzysied - Comments / Approvals plz. The "parent" from this PR has been inflight since Nov 2019.

**Pull request merged**